packages/core/src/connectors/utilities/aliyun.ts

```diff
@@ -0,0 +1,94 @@
+import { createHmac } from 'crypto';
+
+import { has } from '@silverhand/essentials';
+import got from 'got';
+
+import { ConnectorError } from '../types';
+
+// Aliyun has special excape rules.
+// https://help.aliyun.com/document_detail/29442.html
+const escaper = (string_: string) =>
+  encodeURIComponent(string_)
+    .replace(/\*/g, '%2A')
+    .replace(/'/g, '%27')
+    .replace(/!/g, '%21')
+    .replace(/"/g, '%22')
+    .replace(/\(/g, '%28')
+    .replace(/\)/g, '%29')
+    .replace(/\+/, '%2B');
+
+export const getSignature = (
+  parameters: Record<string, string>,
+  secret: string,
+  method: string
+) => {
+  const canoQuery = Object.keys(parameters)
```

Comment:
nit: use `canonicalizedQuery` to make it more clear?
```suggestion
  const canonicalizedQuery = Object.keys(parameters)
```